packages/@aws-cdk/aws-iotevents/lib/DetectorModel.ts

```diff
@@ -0,0 +1,310 @@
+import {
+  IPrincipal,
+  IRole,
+  PolicyStatement,
+  Role,
+  ServicePrincipal,
+} from "@aws-cdk/aws-iam";
+import {
+  Arn,
+  Construct,
+  IResource,
+  Lazy,
+  PhysicalName,
+  Resource,
+  Stack,
+} from "@aws-cdk/core";
+import { CfnDetectorModel } from "./iotevents.generated";
+import { IState } from "./State";
+
+/**
+ * DetectorModel specification
+ *
+ * @export
+ * @interface IDetectorModel
+ * @extends {IResource}
+ * @extends {IGrantable}
+ */
+export interface IDetectorModel extends IResource {
+  /**
+   * ARN for the {@link DetectorModel}
+   *
+   * @type {string}
+   * @memberof IDetectorModel
+   * @attribute
+   */
+  readonly detectorModelArn: string;
+  /**
+   * Name for the {@link DetectorModel}
+   *
+   * @type {string}
+   * @memberof IDetectorModel
+   * @attribute
+   */
+  readonly detectorModelName?: string;
+
+  /**
+   * The IAM service Role for this Detector Model
```

Comment:
describe what the role will be used for  and what permissions is this role expected to have?